### PR TITLE
Use the public transport CA as remote CA if the remote CA list is empty

### PR DESCRIPTION
**Pull request by idanmo**
If remote clusters are not set for an ES cluster, the remote CA `ca.crt` mounted on the ES container is empty, which causes the `_ssl/certificates` ES REST API call to fail. In this PR, if remote clusters are not set for an ES cluster, the remote CA is populated with the public transport CA as a "dummy" CA. 
This fix preserves the behavior where adding/setting remote clusters for ES does not require restarting the ES cluster.

Fixes [#3881](https://github.com/elastic/cloud-on-k8s/issues/3881)

### CODE REVIEW

File: pkg/controller/elasticsearch/certificates/remoteca/reconcile_test.go [hunk 1]
```diff
@@ -101,14 +104,24 @@ func TestReconcile(t *testing.T) {
 						Data: map[string][]byte{certificates.CAFileName: []byte("cert2\n")},
 					},
 				},
+				ca: nil,
```
Comment (pebrc) [inline pkg/controller/elasticsearch/certificates/remoteca/reconcile_test.go:107]: I think we should pass in the `testCA` here and on L64 as well to make sure it does not get included when there are remote CA certs.

**Review by pebrc — APPROVED** [review]
LGTM. I did a quick test and seems to work. The only thing I noticed is that the remote connection is initially disconnected because the CA certificate was not available yet when the remote cluster was set up in ES via the API, but I believe the same could happen before if the CA propagation was too slow.

File: pkg/controller/elasticsearch/certificates/remoteca/reconcile_test.go (hunk 1 above)
Comment (idanmo) [reply pkg/controller/elasticsearch/certificates/remoteca/reconcile_test.go:107]: Yes, that makes sense.

File: pkg/controller/elasticsearch/certificates/remoteca/reconcile.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ func Labels(esName string) client.MatchingLabels {
 func Reconcile(
 	c k8s.Client,
 	es esv1.Elasticsearch,
+	ca *certificates.CA,
```
Comment (sebgl) [inline pkg/controller/elasticsearch/certificates/remoteca/reconcile.go:39]: Let's not make this a pointer so we don't have to check for `nil` in the function.
(In practice we know it's not `nil` in this case (until further refactored :)) but it's not very explicit).

File: pkg/controller/elasticsearch/certificates/remoteca/reconcile.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ func Labels(esName string) client.MatchingLabels {
 func Reconcile(
 	c k8s.Client,
 	es esv1.Elasticsearch,
+	ca certificates.CA,
```
Comment (sebgl) [inline pkg/controller/elasticsearch/certificates/remoteca/reconcile.go:39]: nit: `transportCA` would make things even more explicit

File: pkg/controller/elasticsearch/certificates/remoteca/reconcile_test.go [hunk 4]
```diff
@@ -24,7 +24,9 @@ func TestReconcile(t *testing.T) {
 	type args struct {
 		es      esv1.Elasticsearch
 		secrets []runtime.Object
+		ca      *certificates.CA
```
Comment (sebgl) [inline pkg/controller/elasticsearch/certificates/remoteca/reconcile_test.go:27]: shouldn't be a pointer

**Review by sebgl — APPROVED** [review]
LGTM, I also did a few tests 👍

**Comment by idanmo** [comment]
run full pr build

**Pull request merged**